src/assets/scripts/client/aircraft/AircraftInstanceModel.js

```diff
@@ -2254,6 +1196,51 @@ export default class Aircraft {
         }
     }
 
+    /**
+     * This will display a waring and record an illegal approach event
+     * @for AircraftInstanceModel
+     * @method warnInterceptAngle
+     */
+    warnInterceptAngle() {
+        const isWarning = true;
+        window.uiController.ui_log(`${this.getCallsign()} approach course intercept angle was greater than 30 degrees`, isWarning);
+        window.gameController.events_recordNew(GAME_EVENTS.ILLEGAL_APPROACH_CLEARANCE);
+    }
+
+    /**
+     * This will display update the fix for the aircraft
```

Comment:
Yeah I really had a had time with giving this one a correct name and as well for the comment especially  because it is for  the FIX command where you change the target heading, 

As writing this comment to you I think I found the answer to the strugle